binding/HarfBuzzSharp.Shared/NativeObject.cs

```diff
@@ -66,11 +67,35 @@ internal static IEnumerable<string> PtrToStringArray (IntPtr intPtr)
 			}
 		}
 
-		protected virtual void Dispose (bool disposing)
+		/// <summary>
+		///     Intended to be overridden - always safe to use
+		///     since it will never be called unless applicable
+		/// </summary>
+		protected virtual void DisposeHandler ()
+		{
+		}
+
+		/// <summary>
+		///     Dispose method - always called
+		/// </summary>
+		/// <param name="isDisposing"></param>
+		private void Dispose (bool isDisposing)
 		{
-			if (zero) {
+			if (_isDisposed) {
+				return;
+			}
+
+			if (_zero) {
```

Comment:
This should go after the dispose handler - otherwise the handle will be 0 and then you are going to delete 0.

Reply:
Y makes sense